rust_icu_ucal/src/lib.rs

```diff
@@ -217,4 +331,86 @@ mod tests {
         let tz_version = super::get_tz_data_version().expect("get_tz_data_version works");
         assert!(re.is_match(&tz_version));
     }
+
+    #[test]
+    fn test_get_set_millis() -> Result<(), common::Error> {
+        let now = get_now();
+        let mut cal = UCalendar::new("America/New_York", "en-US", UCalendarType::UCAL_GREGORIAN)?;
+        assert!((now - cal.get_millis()?).abs() < 1000f64);
+
+        let date = now + 17.0;
+        cal.set_millis(date)?;
+        assert_eq!(cal.get_millis()?, date);
+        Ok(())
+    }
+
+    #[test]
+    fn test_set_date() -> Result<(), common::Error> {
+        // Thu May 07 2020 21:00:00 GMT-0400 (US Eastern Daylight Time)
+        let time_a: UDate = 1588899600000.0;
+        // Mon May 04 2020 21:00:00 GMT-0400 (US Eastern Daylight Time)
+        let time_b: UDate = 1588640400000.0;
+
+        let mut cal = UCalendar::new("America/New_York", "en-US", UCalendarType::UCAL_GREGORIAN)?;
+        cal.set_millis(time_a)?;
+        cal.set_date(2020, 4 /* May */, 4)?;
+        assert_eq!(cal.get_millis()?, time_b);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_set_date_time() -> Result<(), common::Error> {
+        // Thu May 07 2020 21:26:55.898 GMT-0400 (US Eastern Daylight Time)
+        let time_a: UDate = 1588901215898.0;
+        // Mon May 04 2020 21:00:00.898 GMT-0400 (US Eastern Daylight Time)
+        let time_b: UDate = 1588640400898.0;
+
+        let mut cal = UCalendar::new("America/New_York", "en-US", UCalendarType::UCAL_GREGORIAN)?;
+        cal.set_millis(time_a)?;
+        cal.set_date_time(2020, 4 /* May */, 4, 21, 0, 0)?;
```

Comment:
Consider also adding a test that is always about a date which is 1 year ahead of whatever is "today".  This way, if things somehow change such that we'd be broken in the future, we'd know 1 year in advance.

Reply:
> Do you mean something like a test that confirms that America/New_York still follows DST one year in the future?

This, in fact.  But, at your option.

Reply:
We don't have any code (to my knowledge) that relies on time zone rules remaining static, so let's avoid creating tests that might break unhelpfully :)